File: ferric_continuum/hello/hello.cc
#include "absl/log/globals.h"
#include "absl/log/initialize.h"
#include "absl/log/log.h"

#include "hello_lib.hh"

int main() {
  // Initialize Abseil logging
  absl::InitializeLog();
  absl::SetStderrThreshold(absl::LogSeverityAtLeast::kInfo);

  LOG(INFO) << ferric::generate_greeting("World");

  // Demonstrate fibonacci
  LOG(INFO) << "Fibonacci sequence (first 10 numbers):";
  for (int i = 0; i < 10; ++i) {
    LOG(INFO) << "fib(" << i << ") = " << ferric::fibonacci(i);
  }

  // Demonstrate prime checking with Abseil string formatting
  LOG(INFO) << "Prime numbers up to 50:";
  auto primes = ferric::primes_up_to(50);
  LOG(INFO) << ferric::format_number_list(primes);

  return 0;
}

File: ferric_continuum/hello/hello.rs
use hello_lib_rs::{fibonacci, generate_greeting, primes_up_to};
use tracing::{info, Level};
use tracing_subscriber;

fn main() {
    // Initialize logging
    tracing_subscriber::fmt().with_max_level(Level::INFO).init();

    info!(greeting = generate_greeting("World"), "Welcome");

    // Demonstrate fibonacci
    info!("Fibonacci sequence (first 10 numbers):");
    for i in 0..10 {
        info!(n = i, result = fibonacci(i), "fib");
    }

    // Demonstrate prime checking
    info!("Prime numbers up to 50:");
    let primes = primes_up_to(50);
    for prime in primes {
        info!(prime, "Found prime");
    }
}
